code/obj/machinery/computer/robotics.dm

```diff
@@ -34,212 +164,74 @@
 	logTheThing(LOG_STATION, src, "is deconstructed by [key_name(user)] at [log_loc(src)]")
 	frame.circuit.id = src.id
 
-/obj/machinery/computer/robotics/process()
-	..()
-	if(status & (NOPOWER|BROKEN))
-		return
-	src.updateDialog()
-	return
+/obj/machinery/computer/robotics/proc/update_silicon_statuses()
+	src.ai_names = list()
+	src.ai_statuses = list()
+	src.ai_killswitch_times = list()
 
+	src.cyborg_names = list()
+	src.cyborg_statuses = list()
+	src.cyborg_cell_charges = list()
+	src.cyborg_modules = list()
+	src.cyborg_lock_times = list()
+	src.cyborg_killswitch_times = list()
```

Comment:
This is a decently janky structure for this. Prefer a list of `cyborg_data` with each of them having whatever info you need to convey about a cyborg. Accessing multiple named params via a hopefully shared index adds unnecessary complexity.